src/librustc/ty/context.rs

```diff
@@ -481,6 +481,8 @@ pub struct GlobalCtxt<'tcx> {
 
     pub maybe_unused_trait_imports: NodeSet,
 
+    pub maybe_unused_extern_crates: Vec<(NodeId, Span, CrateNum)>,
```

Comment:
>I cannot reproduce the pub extern crates case you mentioned

Ah, I see, `pub`s are filtered away in resolve already.

>I'm not sure a visitor is needed.

Okay, not a big deal. I just wanted uniformity between unused imports and extern crates.